src/NATS.Client.Core/NatsConnection.RequestReply.cs

```diff
@@ -61,6 +64,42 @@ public partial class NatsConnection
         }
     }
 
+    [SkipLocalsInit]
+    private static string NewInbox(ReadOnlySpan<char> prefix)
```

Comment:
This method is a bit of a mess. It is rather complicated and still slow and allocates quite heavily. Allocations could be reduced by renting buffers instead of new'ing them, but that reduces throughput.

Maybe it would be possible to not operate with strings (apart from public interfaces) for inbox prefix/subjects and instead use only `byte[]`. But that would be a more invasive change.

Reply:
E.g. this alternative implementation which looks much more reasonable
```csharp
    [SkipLocalsInit]
    private static string NewInbox(ReadOnlySpan<char> prefix)
    {
        Span<char> buffer = stackalloc char[22];
        NuidWriter.TryWriteNuid(buffer);

        if (prefix.IsEmpty)
        {
            return new string(buffer);
        }
        else
        {
            return $"{prefix}.{buffer}";
        }
    }
```

is slower but also reduces the allocations for long prefixes:
```
| Method               | Job           | Runtime       | Mean     | Error    | StdDev   | Ratio | RatioSD | Gen0   | Allocated | Alloc Ratio |
|--------------------- |-------------- |-------------- |---------:|---------:|---------:|------:|--------:|-------:|----------:|------------:|
| NewInbox_ShortPrefix | .NET 6.0      | .NET 6.0      | 79.85 ns | 1.617 ns | 2.370 ns |  0.91 |    0.05 | 0.0612 |     128 B |        1.00 |
| NewInbox_ShortPrefix | .NET 7.0      | .NET 7.0      | 87.93 ns | 1.789 ns | 2.732 ns |  1.00 |    0.00 | 0.0612 |     128 B |        1.00 |
| NewInbox_ShortPrefix | .NET 8.0      | .NET 8.0      | 72.40 ns | 1.353 ns | 1.266 ns |  0.82 |    0.03 | 0.0612 |     128 B |        1.00 |
| NewInbox_ShortPrefix | NativeAOT 8.0 | NativeAOT 8.0 | 66.20 ns | 1.358 ns | 1.270 ns |  0.75 |    0.03 | 0.0612 |     128 B |        1.00 |
|                      |               |               |          |          |          |       |         |        |           |             |
| NewInbox_LongPrefix  | .NET 6.0      | .NET 6.0      | 82.68 ns | 0.958 ns | 1.025 ns |  0.97 |    0.02 | 0.0994 |     208 B |        1.00 |
| NewInbox_LongPrefix  | .NET 7.0      | .NET 7.0      | 85.93 ns | 0.881 ns | 0.688 ns |  1.00 |    0.00 | 0.0994 |     208 B |        1.00 |
| NewInbox_LongPrefix  | .NET 8.0      | .NET 8.0      | 78.37 ns | 1.593 ns | 1.705 ns |  0.91 |    0.02 | 0.0994 |     208 B |        1.00 |
| NewInbox_LongPrefix  | NativeAOT 8.0 | NativeAOT 8.0 | 72.60 ns | 1.492 ns | 1.532 ns |  0.85 |    0.01 | 0.0994 |     208 B |        1.00 |
```